### DISCUSSION TITLE
Docker User in need of assistance

### DISCUSSION THREAD

**Discussion by ShaneBarrett** [abraunegg/onedrive#2814] [Q&A]
Hi,

I use the docker version and watchtower kindly updated to the latest release on the 16th September '24.

But looking at the logs it says that the client version is still 2.4.x and that I should upgrade.... How do I do this? I'm struggling to find the information that will help me.

In addition, the docker version appears to be running, it's trying to sync but throwing up errors and so far today 350GB has been uploaded, but I do not know what it is uploading.
```
ERROR: Encounter std.net.curl.CurlException:

  Error Message:    Stream error in the HTTP/2 framing layer on handle 55D21D6EDD90
  Calling Function: onedrive.OneDriveApi.uploadFragment
  Line number:      683
 Next retry in approx:   0 seconds
Retrying the respective Microsoft Graph API call for Internal Thread ID kFC57B1PqUxIKtT6 (Timestamp: 2024-Sep-17 17:36:50) ..
```
Any help/advice would be gratefully appreciated, I'm thinking it's a me issue but not sure

I love this product btw, keep up the good work!

**Comment by abraunegg** [comment]
@ShaneBarrett 
> But looking at the logs it says that the client version is still 2.4.x and that I should upgrade.... How do I do this? I'm struggling to find the information that will help me.

What logs are indicating this?

You probably have another 'onedrive' client (container or otherwise) running somewhere .....

**Reply by ShaneBarrett** [reply]
The message about client version appears during startup of the container

**Reply by abraunegg** [reply]
Then you have not updated the container correctly and are still running the old Docker container version.

Suggest you purge / delete the old Docker container and pull a new one.

**Comment by abraunegg** [comment]
@ShaneBarrett 
> `Stream error in the HTTP/2 framing layer on handle 55D21D6EDD90`

Regarding this error .. this means that the client you are running has problems with your version of libcurl.

Now ... if you are using Docker, these are the versions of Linux being used:

* Dockerfile = Fedora 40
* Dockerfile-alpine = Alpine 3.20
* Dockerfile-debian = Debian 12

Now ... what version of 'curl' is this using:

* Fedora 40 = 8.6
* Alpine 3.20 = 8.9x
* Debian 12 = 7.88

Now ... please read this: https://github.com/abraunegg/onedrive/blob/master/docs/usage.md#compatibility-with-curl

![image](https://github.com/user-attachments/assets/d50a4fff-fcef-48a5-8348-c46fef5295b0)

So .. most likely you are using the Debian Docker container, which, has known HTTP2 problems due the version of 'curl' that Debian provides.

Your options are:
* Raise a bug with the Debian team to upgrade their 'curl' version to something more modern - and good luck with that.
* Configure your Docker container to use HTTP/1.1 protocol - please read the documentation on this.
* Use either the Fedora or Alpine Docker container and stop using the Debian one

**Reply by ShaneBarrett** [reply]
You are correct, I am running the Debian docker version. driveone/onedrive:edge

Forgive my ignorance but is it possible that I could run driveone/onedrive:fedora  even though I run Debian as my OS for my home lab?

**Reply by abraunegg** [reply]
Thats the point of Docker to allow that.

If you enjoy running a distro that gives you old packages with bugs in the guise of stability - keep running Debian or rethink your home lab and use something more modern

**Reply by ShaneBarrett** [reply]
Thank you, I will let you know how I get on

**Reply by ShaneBarrett** [reply]
Setup a new docker container from scratch, but there seems to be a problem...

Apologies in advanced for the formatting
```
usermod: no changes
# Running container as user: onedrive
# Base Args: --monitor
# Changing ownership permissions on /onedrive/data and /onedrive/conf to onedrive:users
# Launching 'onedrive' as onedrive via gosu
Reading configuration file: /onedrive/conf/config
Configuration file successfully loaded
Application configuration file has been updated, checking if --resync needed
WARNING: no backup config file was found, unable to validate if any changes made
Using IPv4 and IPv6 (if configured) for all network operations
Attempting to contact Microsoft OneDrive Login Service
Successfully reached Microsoft OneDrive Login Service
Configuring Global Azure AD Endpoints
OneDrive synchronisation interval (seconds): 300
Initialising filesystem inotify monitoring ...
Performing initial synchronisation to ensure consistent local state ...
Attempting to contact Microsoft OneDrive Login Service
Successfully reached Microsoft OneDrive Login Service
Starting a sync with Microsoft OneDrive
Fetching items from the OneDrive API for Drive ID: REDACTED ...................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................... 
Processing 327474 applicable changes and items received from Microsoft OneDrive ..................................................................................................................................................................................................................................................................................................... 
Number of items to download from Microsoft OneDrive: 8
Due to a termination signal, internal processing stopped abruptly. The application will now exit in a unclean manner.
Due to a termination signal, internal processing stopped abruptly. The application will now exit in a unclean manner.
Due to a termination signal, internal processing stopped abruptly. The application will now exit in a unclean manner.
Due to a termination signal, internal processing stopped abruptly. The application will now exit in a unclean manner.
Due to a termination signal, internal processing stopped abruptly. The application will now exit in a unclean manner.
Due to a termination signal, internal processing stopped abruptly. The application will now exit in a unclean manner.
Due to a termination signal, internal processing stopped abruptly. The application will now exit in a unclean manner.
Due to a termination signal, internal processing stopped abruptly. The application will now exit in a unclean manner.
```

I'm using image driveone/onedrive:edge-alpine

**Comment by abraunegg** [comment]
@ShaneBarrett 
Please generate a debug log as per: https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support#docker-container-example

**Reply by ShaneBarrett** [reply]
OK, this may take a little while as there is about 500GB of data... will e-mail a link to the log file when it is done

**Reply by ShaneBarrett** [reply]
Email sent with link to log file

**Reply by ShaneBarrett** [reply]
Out of curiosity more than anything, I did a new instance using fedora-edge and so far it seems to be behaving 

I did the exact same steps as I did for alpine-edge

**Reply by abraunegg** [reply]
> Out of curiosity more than anything, I did a new instance using fedora-edge and so far it seems to be behaving

Interesting ... Potentially a incompatibility with Alpine and Debian (which is also not surprising)

Things for you to also try:

1. Install Docker the *correct* way - using the Docker packages - not the Debian ones
2. Move your entire platform away from Debian or Ubuntu based due to all the old packages they use in the guise of 'stability'

**Reply by ShaneBarrett** [reply]
Thanks, I will see how this fedora version goes, so far it is working exactly as expected! 

This update is awesome thank you for the hard work you put into it!

If I have issues further down the line, I may look to move away from a Debian base

**Comment by abraunegg** [comment]
@ShaneBarrett 
FYI, the Debian Docker images have now been updated with a newer version of 'curl' to avoid running into the issues like `Stream error in the HTTP/2 framing layer on handle 55D21D6EDD90` presented by the default version of 'curl' in Debian.

If you are using the 'edge' Docker tag, once these are rebuilt, you will get that fix.
